ChatSecure/Classes/Controllers/XMPP/OTRStreamManagementYapStorage.m

```diff
@@ -0,0 +1,312 @@
+//
+//  OTRStreamManagementYapStorage.m
+//  ChatSecure
+//
+//  Created by David Chiles on 11/19/14.
+//  Copyright (c) 2014 Chris Ballinger. All rights reserved.
+//
+
+#import "OTRStreamManagementYapStorage.h"
+#import "XMPPStreamManagement.h"
+#import "OTRDatabaseManager.h"
+#import "OTRStreamManagementStorageObject.h"
+
+@interface OTRStreamManagementYapStorage ()
+
+@property (nonatomic, strong) YapDatabaseConnection *databaseConnection;
+@property (nonatomic) dispatch_queue_t parentQueue;
+
+@end
+
+@implementation OTRStreamManagementYapStorage
+
+- (instancetype)init
+{
+    if (self = [super init]) {
+        self.databaseConnection = [[OTRDatabaseManager sharedInstance] newConnection];
```

Comment:
@davidchiles Is it possible to somehow pass in the databaseConnection as an initialization parameter, or a set property?